packages/react-storage/src/components/StorageBrowser/actions/handlers/listLocationsAction.ts

```diff
@@ -0,0 +1,42 @@
+import { Permission } from '@aws-amplify/storage/internals';
+
+import {
+  ActionInputConfig,
+  ListActionOptions,
+  ListActionInput,
+  ListActionOutput,
+  ListAction,
+  LocationType,
+} from '../types';
+
+export interface LocationData<T = Permission, K = LocationType> {
+  bucket: string;
+  permission: T;
+  prefix: string;
+  type: K;
+}
+
+export interface ListLocationsActionOptions<T = Permission | LocationType>
+  extends ListActionOptions {
+  exclude?: T | T[];
```

Comment:
Suggestion: Separating these would likely increase type safety and clarity. Maybe something like -

```
excludePermissions?: Permission | Permission[];
excludeLocationTypes?: LocationType | LocationType[];
```

Reply:
Preference is to keep the API surface layer minimal here and IMO a singular `exclude` is expressive enough that multiple keys are not required. Am curious as to why type safety would be impacted since `Permission` and `LocationType` are string unions?

For some context, original intent here was to allow for removal of the excluded keys from the return type but not sure if that is something actually needed so can probably drop the generic

Reply:
I wasn't sure what `Permission` was, but this is okay if they're both just string unions. I think we can avoid using generics unless we really need it

Reply:
I kind of like having the specific key versus generic here. Is it possible we want to exclude both Permission and Location type in the future? Also can we always expect array to simplify the interface?

Reply:
Forcing an array as the shape of `exclude` has a negative impact on DX for consumers that only want to filter out a single key. Additionally, `ListLocationItemsAction` needs to support exclusion of `FILE` and `FOLDER`, which wouldn't make sense in the array form

Reply:
+1, I think allowing T or T[] makes sense here rather than forcing users to always provide an array when they only want to filter a single key.